torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -470,54 +470,44 @@ struct TraceState {
   std::vector<size_t> output_grad_targets;
 };
 
-#define SWAP_SAVED_VARIABLES_SAVE(mapping, var, move) \
-  bool inserted = mapping.emplace(&var, move).second; \
-  TORCH_INTERNAL_ASSERT(inserted, "duplicate before()");
-
-#define SWAP_SAVED_VARIABLES_RESTORE(mapping, var)                 \
-  auto it = mapping.find(&var);                                    \
-  TORCH_INTERNAL_ASSERT(it != mapping.end(), "duplicate after()"); \
-  var = std::move(it->second);                                     \
-  mapping.erase(it);
-
 class SwapSavedVariables {
   // SwapSavedVariables is used during the tracing/compilation phase after a
   // cache-miss. It swaps any 'lifted' inputs (tensors, symints) to proxy nodes,
   // allows tracing to happen, then swaps them back afterwards.
  public:
   void before(at::Tensor& t) {
     TensorArg& arg = compiler.tensor_args.lookup(t);
-    SWAP_SAVED_VARIABLES_SAVE(stashed_tensors, t, std::move(t));
+    stashed_tensors.save(&t, std::move(t));
     if (arg.defined()) {
       TORCH_INTERNAL_ASSERT(arg.proxy_tensor.defined());
       t = arg.proxy_tensor;
     }
   }
   void after(at::Tensor& t) {
-    SWAP_SAVED_VARIABLES_RESTORE(stashed_tensors, t);
+    stashed_tensors.restore(&t);
   }
 
   void before(SavedVariable& t) {
     TensorArg& arg = compiler.tensor_args.lookup(t);
-    SWAP_SAVED_VARIABLES_SAVE(stashed_variables, t, std::move(t));
+    stashed_variables.save(&t, std::move(t));
     if (arg.defined()) {
       TORCH_INTERNAL_ASSERT(arg.proxy_tensor.defined());
       t = SavedVariable(arg.proxy_tensor, false);
     }
   }
   void after(SavedVariable& t) {
-    SWAP_SAVED_VARIABLES_RESTORE(stashed_variables, t);
+    stashed_variables.restore(&t);
   }
 
   void before(c10::SymInt& t) {
-    SWAP_SAVED_VARIABLES_SAVE(stashed_symints, t, t);
+    stashed_symints.save(&t, c10::SymInt(t));
```

Comment:
Why are we copying the SymInt here?

Reply:
`save()` requires either `std::move()` or copy due to the `&&` arg.  We need the std::move() in the other cases because SavedVariable doesn't support copy construction, but don't want it here.